tests/test_unordered.py

```diff
@@ -7,20 +7,85 @@
 
 
 @pytest.mark.parametrize(
-    "left,right",
+    ["expected", "actual"],
     [
         (unordered(1, 2, 3), [3, 2, 1]),
-        ([3, 2, 1], unordered(1, 2, 3)),
+        (unordered(1, 2, 3), (3, 2, 1)),
+        (unordered(1, 2, 3), {3, 2, 1}),
+        (unordered([1, 2, 3]), [3, 2, 1]),
+        (unordered((1, 2, 3)), (3, 2, 1)),
+        (unordered({1, 2, 3}), {3, 2, 1}),
+        (unordered({1: 2, 3: 4}), {3: 4, 1: 2}),
```

Comment:
This is ambiguous. What is an unordered `dict`?

Reply:
I added this because `Unordered`  expects an `Iterable` and as `dict` is also an `Iterable` I added this less obvious example. In practice it indeed does not make a lot of sense. You want to drop it?

Reply:
I don't want to drop this line, but I want to present least surprise to the users. Currently, `unordered({1: 2, 3: 4}) == {3: 5, 1: 6}` will also hold, which the users don't expect.

Reply:
oh yes, I didn't think of that. Maybe it's best to reject mapping types in unordered then to avoid accidental surprise?

Reply:
It is a good idea. I was thinking about keeping the old behaviour for dicts, i.e. `unordered({1: 2, 3: 4}) == [{1: 2, 3: 4}]`, but this would not be in line with your changes.

Reply:
> unordered({1: 2, 3: 4}) == [{1: 2, 3: 4}]

this looks weird to me to be honest, I guess you see `unordered` as a factory for a new object (a list that does not care about order), while I see it more as a wrapper around an existing sequence (to ignore order when checking equality).

What do you propose to resolve this? Do I remove the test case to finish up this PR and leave the dict handling to another issue?

Reply:
It should be resolved here. Please implement your idea: rejecting mapping types in `unordered`.

Reply:
ok done

Reply:
Thank you!